tests/conftest.py

```diff
@@ -914,12 +918,58 @@ def creds_all_duts(duthosts):
     return creds_all_duts
 
 
+def log_custom_msg(item):
+    # temp log output to track module name
+    logger.debug("[log_custom_msg] item: {}".format(item))
+    customMsgDict = {}
+    dutChekResults = {}
+    for key in DUT_CHEK_LIST:
+        if item.session.config.cache.get(key, None):
+            dutChekResults[key] = item.session.config.cache.get(key, None)
+            item.session.config.cache.set(key, None)
+    if dutChekResults:
+        customMsgDict['DutChekResult'] = dutChekResults
+    # Check pre_sanity_checks results
+    preSanityCheckResults = {}
+    if item.session.config.cache.get("pre_sanity_check_failed", None):
```

Comment:
This logic is not extendable, can we change the logic of cache selection from the magic code to auto detection?
"post_sanity_check_failed" -> "sonic_custom_msg.dut_check_result.post_sanity_check_failed"
"sonic_custom_msg" means this cache should be caught in to CustomMsg
"dut_check_result.post_sanity_check_failed" means finally it would be 
```
{
  "dut_check_result": {"post_sanity_check_failed": val}
} 
```

And we don't use 'PreSanityCheckResults' and 'PostSanityCheckResults' as the parent,
let's include all sanity check result and coore dump check result into "dut_check_result".